sdk/core/AzureCore/Source/Pipeline/PipelineClient.swift

```diff
@@ -97,7 +97,7 @@ open class PipelineClient {
                 let allowedStatusCodes = pipelineResponse.value(forKey: .allowedStatusCodes) as? [Int] ?? [200]
                 if !allowedStatusCodes.contains(httpResponse?.statusCode ?? -1) {
                     self.logError(withData: deserializedData)
-                    let error = HTTPResponseError.statusCode("Service returned invalid status code [\(statusCode)].")
+                    let error = AzureError.service("Service returned invalid status code [\(statusCode)].", nil)
```

Comment:
This is probably a `.sdk` - our SDK is defining what the allowed status codes should be.

Reply:
The Swagger defines what the allowed status codes can be.

Reply:
Fair enough, I don't feel strongly.